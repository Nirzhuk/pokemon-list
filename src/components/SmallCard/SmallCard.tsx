import React from 'react'
import styled from 'styled-components'

const SmallCardContainer = styled.div`
    margin: 10px;
    padding: 8px 12px;
    text-align: center;
    border: 1px solid #202020;
    border-radius: 8px;

    &:hover {
        border-color: #444444;
        background: #e2e2e2;
        ${({ onClick }) => onClick && `cursor: pointer;`}
    }
`;

export interface SmallCardProps {
    text: string;
    onClick?: ()=> void;
}

const SmallCard = ({ text, onClick }: SmallCardProps) => <SmallCardContainer onClick={onClick}>{text}</SmallCardContainer>

export default SmallCard