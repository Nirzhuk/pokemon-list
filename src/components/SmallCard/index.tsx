export { default, type SmallCardProps } from './SmallCard'