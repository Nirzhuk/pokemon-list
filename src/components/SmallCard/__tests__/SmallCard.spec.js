import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import SmallCard from '../SmallCard'

describe('SmallCard', () => {
    const props = {
        text: 'TextSmallCard'
    }
    const setup = (initialProps = props) => render(<SmallCard {...initialProps}/>)
    it('Renders correctly', () => {
        const { container } = setup();
        expect(screen.getByText('TextSmallCard')).toBeInTheDocument()
        expect(container).toMatchSnapshot();
    })

    it('Renders correctly',async  () => {
        const handleClick = jest.fn()
        setup({ text: 'New text', onClick: handleClick });

        await userEvent.click(screen.getByText('New text'))

        expect(handleClick).toHaveBeenCalledTimes(1);
    })
})