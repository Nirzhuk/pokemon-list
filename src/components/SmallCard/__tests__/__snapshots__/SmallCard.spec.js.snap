// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallCard Renders correctly 1`] = `
<div>
  <div
    class="sc-bcXHqe lbydtb"
  >
    TextSmallCard
  </div>
</div>
`;
